src/mongocxx/private/libmongoc_symbols.hh

```diff
@@ -29,7 +31,9 @@ MONGOCXX_LIBMONGOC_SYMBOL(bulk_operation_update_many_with_opts)
 MONGOCXX_LIBMONGOC_SYMBOL(bulk_operation_update_one_with_opts)
 MONGOCXX_LIBMONGOC_SYMBOL(cleanup)
 MONGOCXX_LIBMONGOC_SYMBOL(client_destroy)
+BSONCXX_SUPPRESS_DEPRECATION_WARNINGS_BEGIN
 MONGOCXX_LIBMONGOC_SYMBOL(client_find_databases)
```

Comment:
this is just a nitpick, but since there are just a few of these would it make more sense just switch over to the `with_opts` variants now?

Reply:
I considered it, and in fact started down that path. However, the error handling model for the `with_opts` is different than that for the old variants - they aren't just drop-in replacements. It seemed better to do that for 3.3.0.

Also, this fix may be incomplete. On older compilers, we still seem to get deprecation warnings, which get promoted to errors since we build with -Werror. I think for some reason older GCC is evaluating the call through the function pointer to the deprecated function as being a usage of a deprecated thing.

https://evergreen.mongodb.com/task_log_raw/cxx_driver_ubuntu1404_debug_compile_and_test_with_shared_libs_patch_fc9a44325e03bb8bb166e814caeb23413b4ba7af_5a5a51e7e3c3316075002589_18_01_13_18_37_41/0?type=T#L1312

I'll probably just disable -Werror on that builder, but expect another update to this PR soon.